Some code style fixes

roji opened the pull request:
While going over the NTS source code, I encountered several coding style issues - modern C# 7 features aren't used (e.g. expression-bodied properties and methods), indentation and newlines are frequently inconsistent...

This is a first PR that addresses some of these issues. If you're interested I can help out a bit more on cleanup.

(that's what happens when you're an obssessive-compulsive personality...)

FObermaier commented:
Travis-CI seems to dislike Expression-bodied properties, at least the way it is configured.
Doesn't that prevent all VS2015 users work on the code?

airbreather commented on NetTopologySuite/Algorithm/CentroidArea.cs:
```diff
@@ -177,14 +177,12 @@ private static void Centroid3(Coordinate p1, Coordinate p2, Coordinate p3, ref C
-        {
-            return (p2.X - p1.X) * (p3.Y - p1.Y) - (p3.X - p1.X) * (p2.Y - p1.Y);
-        }
+            s=> (p2.X - p1.X) * (p3.Y - p1.Y) - (p3.X - p1.X) * (p2.Y - p1.Y);
```
syntax error

airbreather commented:
@FObermaier I did a quick "review" to show the line that Travis is probably complaining about

roji commented:
Sorry for wasting your time and pushing that, it contained a few errors. I've cleaned things up and verified that everything builds fine, hopefully your Travis build should complete now. I've also rebased this PR on the latest develop so merging should be OK without conflicts.

@FObermaier yes, some of the changes make use of C# 7 language features, which are available from Visual Studio 2017 and above. This only affects people trying to *build* NTS, though, and not people using it. IMHO at this point it's quite reasonable use C# 7 and to expect developers to work with it... do you think otherwise?

roji commented:
OK, the Travis build is green now.

Note that almost all of the changes here (save the last commit for expression-bodied methods) were done automatically with Project Rider, which should mean that they're safe - but if could be a good idea to quickly scan through and make sure everything makes sense (I did this as well).

FObermaier commented:
@roji Project Rider == JetBrains' Rider open source IDE?

roji commented:
Yep - although I don't think it's open source... It's basically their Java IDE platform (used for Python PyCharm, IntelliJ Java...) With the Resharper engine (which is used for refactorings such as these).

Works amazingly well, especially since I'm a Linux guy...

FObermaier commented:
I'm going to merge [PR229](https://github.com/NetTopologySuite/NetTopologySuite/pull/228) first. Maybe it is easier to redo the code inspections based on the result, as a lot of files are moved to seperate repositories? At least I assume it makes a rebase difficult.

roji commented:
No problem, I'll take a look at rebasing this after the merging of #229.

roji commented:
Closing for now, it's better to first normalize line endings (#230), once that's merged I'll resubmit.
